flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -128,6 +128,13 @@ else if (message instanceof Status.Success) {
 			// job was successfully submitted :-)
 			logger.info("Job was successfully submitted to the JobManager");
 		}
+		else if (message instanceof JobClientMessages.JobFailure) {
```

Comment:
Wrapping the `SerializedThrowable` in a `Failure` would allow us to get rid off this if branch.